src/components/select/Select.svelte

```diff
@@ -0,0 +1,864 @@
+    } else if (formerValue !== value) {
+      console.log("changed", formerValue, value);
+      dispatch("change", { value, label: getSelectedLabel() });
+    }
+  }
+  // use a function to fetch the label so the above
+  // watcher doesn't fire on every change to the label
+  function getSelectedLabel() {
+    return $selectStore.selectedLabel;
+  }
+
+  $: if (!$selectStore.optionsVisible) {
+    // Allow transition to finish before clearing search
+    setTimeout(() => {
+      if ($selectStore.searchValue && autoClearSearchValue) {
+        $selectStore.searchValue = "";
+      }
+    }, 200);
+  }
+
+  // When the search value changes, we need to know when
+  // there are no options so that we can display an empty message
+  $: if (typeof $selectStore.searchValue) {
+    $selectStore.allOptionNodes =
+      $selectStore.dropdownNode &&
+      $selectStore.dropdownNode.querySelectorAll(`.${prefixCls}-item-option`);
+  }
+
+  $: noOptions =
+    $selectStore.allOptionNodes &&
+    !$selectStore.allOptionNodes.length &&
+    !$selectStore.pendingTag &&
+    !$selectStore.addedTags.length;
+
+  $: showPlaceholder = (function() {
+    if (!placeholder) return false;
+    if (
+      !isSingleMode &&
+      $selectStore.selectedValue &&
+      !$selectStore.selectedValue.length
+    ) {
+      return true;
+    }
+    if (!$selectStore.selectedValue) return true;
+  })();
+
+  // Set input width as the user types to allow maximum space for selected options
+  $: if (
+    typeof document !== "undefined" &&
+    document.getElementById(wrapperId)
+  ) {
+    const item = document
+      .getElementById(wrapperId)
+      .querySelector(`.${prefixCls}`);
+    const size = calculateSize($selectStore.searchValue, {
+      font: "Arial",
+      fontSize: window.getComputedStyle(item).fontSize
+    });
+    inputWidth = `${size.width + $selectStore.searchValue.length}px`;
+  }
+
+  // In multiple mode, make sure focus stays on
+  // the input even after selecting/deselecting options
+  $: if (
+    typeof document !== "undefined" &&
+    !isSingleMode &&
+    $selectStore.selectedValue &&
+    $selectStore.optionsVisible
+  ) {
+    inputNode.focus();
+  }
+
+  // A user can use the tagProps option to pass a function
+  // that will render tags with custom props
+  $: if (typeof $selectStore.selectedValue === "object" && tagProps) {
+    currentTagAttributes = $selectStore.selectedValue.map((value, index) => {
+      const props = tagProps({
+        value,
+        label: $selectStore.selectedLabel[index]
+      });
+
+      // So we can saftely delete values without worrying
+      // about references when we want to call a function
+      const propsCopy = { ...props };
+
+      let onClose = () => {};
+      if (props.onClose) {
+        onClose = event => {
+          propsCopy.onClose(event);
+          removeOption(index);
+        };
+        delete props.onClose;
+      }
+
+      let onClick = () => {};
+      if (props.onClick) {
```

Comment:
Why are we passing `onClick` as a param here instead of using an event?

Reply:
The user can pass `tagProps` to the Select and if they do they don't have direct access to add events on the Tag. This gives them the ability to pass a click function to the Select and the Select will add the `on:click` event here: https://github.com/svelte-components/svant/blob/aaronsaloff/select_component/src/components/select/Select.svelte#L17

Reply:
Makes sense.